bot/exts/recruitment/talentpool/_review.py

```diff
@@ -44,37 +50,111 @@
+            if is_first_message:
+                time_since_message_created = datetime.now(timezone.utc) - msg.created_at
+                if time_since_message_created < MIN_REVIEW_INTERVAL:
+                    log.debug("Most recent review was less than %s ago, cancelling check", MIN_REVIEW_INTERVAL)
+                    return False
+
+                is_first_message = False
+
+            review_count += 1
+
+            if review_count >= MAX_ONGOING_REVIEWS:
+                log.debug("There are already at least %s ongoing reviews, cancelling check.", MAX_ONGOING_REVIEWS)
+                return False
+
+        return True
 
+    async def get_user_for_review(self) -> Optional[int]:
+        """
+        Returns the user ID of the next user to review, or None if there are no users ready.
+
+        Users will only be selected for review if:
+         - They have not already been reviewed.
+         - They have been nominated for longer than `MIN_NOMINATION_TIME`.
+
+        The priority of the review is determined by how many nominations the user has
+        (more nominations = higher priority).
+        For users with equal priority the oldest nomination will be reviewed first.
+        """
+        possible = []
         for user_id, user_data in self._pool.cache.items():
-            if not user_data["reviewed"]:
-                self.schedule_review(user_id)
+            time_since_nomination = datetime.now(timezone.utc) - isoparse(user_data["inserted_at"])
+            if (
+                not user_data["reviewed"]
+                and time_since_nomination > MIN_NOMINATION_TIME
+            ):
+                possible.append((user_id, user_data))
+
+        if not possible:
+            log.debug("No users ready to review.")
+            return None
+
+        oldest_date = min(isoparse(x[1]["inserted_at"]) for x in possible)
+        max_entries = max(len(x[1]["entries"]) for x in possible)
+
+        def sort_key(nomination: dict) -> float:
+            return self.score_nomination(nomination[1], oldest_date, max_entries)
+
+        return max(possible, key=sort_key)[0]
+
+    @staticmethod
+    def score_nomination(nomination: dict, oldest_date: datetime, max_entries: int) -> float:
+        """
+        Scores a nomination based on age and number of nomination entries.
 
-    def schedule_review(self, user_id: int) -> None:
-        """Schedules a single user for review."""
-        log.trace(f"Scheduling review of user with ID {user_id}")
+        The higher the score, the higher the priority for being put up for review should be.
+        """
+        num_entries = len(nomination["entries"])
+        entries_score = num_entries / max_entries
 
-        user_data = self._pool.cache.get(user_id)
-        inserted_at = isoparse(user_data['inserted_at'])
-        review_at = inserted_at + timedelta(days=MAX_DAYS_IN_POOL)
+        nomination_date = isoparse(nomination["inserted_at"])
+        now = datetime.now(timezone.utc)
```

Comment:
It's not critical in practice, because of the speed of evaluation and the difference between nomination dates, but `now` should be constant across the nominations for correct behavior.

Reply:
True, fixed 👍